Various updates for 6.3.0

pprindeville opened the pull request:
The RFC-7895 module library is no longer present in 6.3.0.

olofhagsand reviewed (commented):
The www-data is a remnant of  running restconf with nginx as a reverse proxy, which many upstream users still do, and where www-data is the natural choice for the restconf daemon. For native restconf, the clicon user is natural.
The way to choose between nginx and native is the configure option: --with-restconf=native vs --with-restconf=fcgi. 
To make this choice automatic, one could use this factin the .xml.in to make the selection.

olofhagsand commented on configure.ac:
```diff
@@ -147,15 +147,19 @@ AC_ARG_ENABLE(nls)
 AC_OUTPUT([Makefile
 	  hello/Makefile
   	  hello/src/Makefile
+  	  hello/src/hello.xm
```
Typo: should be hello/src/hello.xml

pprindeville replied:
Done

olofhagsand reviewed (commented):
There is also an issue with building the dockers in the upstream CI that stem from commit https://github.com/clicon/clixon/commit/989aa254283d5b3f29245da23418fcb9d95a5c9f.
I will try to fix that separetely.
Also, this PR seems to have swollen outside the initial scope.
Maybe change the title or break it up into smaller bits?

pprindeville commented:
> There is also an issue with building the dockers in the upstream CI that stem from commit [clicon/clixon@989aa25](https://github.com/clicon/clixon/commit/989aa254283d5b3f29245da23418fcb9d95a5c9f). A fix on this has been pushed separately, please rebase. Also, this PR seems to have swollen outside the initial scope. 

Rebased.

> Maybe change the title or break it up into smaller bits?

Changed the title.

> Finally, I see these warnings:
> 
> ```
> config.status: WARNING:  'hello/src/hello.xml.in' seems to ignore the --datarootdir setting
> config.status: WARNING:  'wifi/src/wifi.xml.in' seems to ignore the --datarootdir setting
> config.status: WARNING:  'openconfig/src/openconfig.xml.in' seems to ignore the --datarootdir setting
> config.status: WARNING:  '3gpp/src/3gpp.xml.in' seems to ignore the --datarootdir setting
> ```

Those are harmless.

pprindeville commented:
> The www-data is a remnant of running restconf with nginx as a reverse proxy, which many upstream users still do, and where www-data is the natural choice for the restconf daemon. For native restconf, the clicon user is natural.
> The way to choose between nginx and native is the configure option: --with-restconf=native vs --with-restconf=fcgi.
> To make this choice automatic, one could use this factin the .xml.in to make the selection.

What would that look like?

pprindeville commented:
> The www-data is a remnant of running restconf with nginx as a reverse proxy, which many upstream users still do, and where www-data is the natural choice for the restconf daemon. For native restconf, the clicon user is natural.
> The way to choose between nginx and native is the configure option: --with-restconf=native vs --with-restconf=fcgi.
> To make this choice automatic, one could use this factin the .xml.in to make the selection.

And PR #442 isn't adequate?

olofhagsand commented:
Thanks for bringing the examples up to date
